src/AppBundle/Resources/config/services.yml

```diff
@@ -23,17 +23,19 @@ services:
         tags:
             - { name: ezpublish.query_type }
 
-    app.criteria.menu:
-        class: AppBundle\Criteria\Menu
+    app.query_type.menu:
+        class: AppBundle\QueryType\MenuQueryType
+        calls:
+            - [setLanguages, [$languages$]]
+        tags:
+            - { name: ezpublish.query_type }
 
     app.controller.menu:
         class: AppBundle\Controller\MenuController
         arguments:
             - '@templating'
             - '@ezpublish.api.service.search'
-            - '@ezpublish.api.service.location'
-            - '@ezpublish.config.resolver'
-            - '@app.criteria.menu'
+            - '@ezpublish.query_type.registry'
```

Comment:
Yes, I know. It's only for demonstration purpose, look: https://github.com/ezsystems/ezplatform-demo/pull/39/files/92940b0eeea2f26caa09207a209a1307b19a2807#diff-71435085ac34f5c4748063a144167686R42